cltk/tokenize/sentence.py

```diff
@@ -109,10 +135,113 @@ def indian_punctuation_tokenize_regex(self: object, untokenized_string: str):
         return re.sub(r'[ ]+',u' ',tok_str).strip(' ').split(' ')
 
     def tokenize(self: object, untokenized_string: str):
-        # NLTK's PlaintextCorpusReader needs a function called tokenize
-        # in functions used as a parameter for sentence tokenization.
-        # So this is an alias for tokenize_sentences().
+        """Alias for tokenize_sentences()—NLTK's PlaintextCorpusReader needs a
+        function called tokenize in functions used as a parameter for sentence
+        tokenization.
+
+        :type untokenized_string: str
+        :param untokenized_string: A string containing one of more sentences.
+        """
         if self.language in INDIAN_LANGUAGES:
             return self.indian_punctuation_tokenize_regex(untokenized_string)
         else:
             return self.tokenize_sentences(untokenized_string)
+
+### Code below for consideration as new structure for modules; code above legacy?
+
+from abc import abstractmethod
+
+class BaseSentenceTokenizer(object):
+    """ Base class for sentence tokenization
+    """
+
+    def __init__(self: object, language: str = None):
+        """ Initialize stoplist builder with option for language specific parameters
+        :param language : language for sentence tokenization
+        :type language: str
+        """
+        if language:
+            self.language = language.lower()
+
+    @abstractmethod
+    def tokenize(self: object, text: str):
+        """
+        Method for tokenizing sentences. This method
+        should be overridden by subclasses of SentenceTokenizer.
+        """
+
+
+class BasePunktSentenceTokenizer(BaseSentenceTokenizer):
+    """Base class for punkt sentence tokenization
+    """
+
+    missing_models_message = "BasePunktSentenceTokenizer requires a language model."
+
+    def __init__(self: object, language: str = None, lang_vars: object = None):
+        """
+        :param language : language for sentence tokenization
+        :type language: str
+        """
+        self.language = language
+        self.lang_vars = lang_vars
+        BaseSentenceTokenizer.__init__(self, language=self.language)
```

Comment:
prefer PY3 super style.

Reply:
@todd-cook This (see above about the workaround) is not working—gets an error: ```TypeError: super(type, obj): obj must be an instance or subtype of type```. Thoughts?